.github/workflows/cmake.yml

```
@@ -0,0 +1,89 @@
+# inspired by "build GDAL library and to run test with CMake"
+# from https://github.com/OSGeo/gdal/blob/master/.github/workflows/cmake_builds.yml
+
+---
+name: CMake
+
+on:
+    push:
+        paths-ignore:
+            - 'doc/**'
+        branches-ignore:
+            - 'backport**'
+    pull_request:
+        paths-ignore:
+            - 'doc/**'
+
+concurrency:
+    group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+    cancel-in-progress: true
+
+env:
+    CMAKE_UNITY_BUILD: OFF
+
+permissions:
+    contents: read  #  to fetch code (actions/checkout)
+
+jobs:
+
+    build-cmake:
+        runs-on: ubuntu-20.04
+        env:
+            CMAKE_OPTIONS: -DPython_LOOKUP_VERSION=3.8 -DUSE_CCACHE=ON
+            cache-name: cmake-ubuntu-20.04
+        steps:
+            - name: Checkout GRASS
+              uses: actions/checkout@v4.1.1
+            - name: Setup cache
+              uses: actions/cache@v4.0.2
+              id: cache
+              with:
+                  path: ${{ github.workspace }}/.ccache
+                  key: "${{ env.cache-name }}-${{ runner.os }}-${{ github.base_ref }}\
+                        ${{ github.ref_name }}-${{ github.run_id }}"
+                  restore-keys: |
+                    ${{ env.cache-name }}-${{ runner.os }}-${{ github.base_ref }}
+                    ${{ env.cache-name }}-${{ runner.os }}
+            - name: Install CMake 3.16.0
+              run: |
+                cd ${GITHUB_WORKSPACE}
+                wget https://github.com/Kitware/CMake/releases/download/v3.16.0/cmake-3.16.0-Linux-x86_64.tar.gz
+                tar xzf cmake-3.16.0-Linux-x86_64.tar.gz
+                echo "CMAKE_DIR=$GITHUB_WORKSPACE/cmake-3.16.0-Linux-x86_64/bin" >> $GITHUB_ENV
+                echo "$GITHUB_WORKSPACE/cmake-3.16.0-Linux-x86_64/bin" >> $GITHUB_PATH
+            - run: |
+                cmake --version
+            - name: Install dependencies
+              run: |
+                sudo add-apt-repository -y ppa:ubuntugis/ubuntugis-unstable
+                sudo apt-get update
+                sudo apt-get install -y -q build-essential \
+                 flex make bison gcc libgcc1 g++ ccache \
+                 python3 python3-dev \
+                 python3-opengl python3-wxgtk4.0 \
+                 python3-dateutil libgsl-dev python3-numpy \
+                 wx3.0-headers wx-common libwxgtk3.0-gtk3-dev \
+                 libwxbase3.0-dev libbz2-dev zlib1g-dev gettext \
+                 libtiff5-dev libpnglite-dev libcairo2 libcairo2-dev \
+                 sqlite3 libsqlite3-dev \
+                 libpq-dev libreadline6-dev libfreetype6-dev \
+                 libfftw3-3 libfftw3-dev \
+                 libboost-thread-dev libboost-program-options-dev libpdal-dev\
+                 subversion libzstd-dev checkinstall \
+                 libglu1-mesa-dev libxmu-dev \
+                 ghostscript wget -y make ccache g++
+            - name: Configure ccache
```

```suggestion
            - run: python -m pip install ninja
            - name: Configure ccache
```
Make a recent version of ninja available. Chose pip rather than apt-get as it is outdated on the old ubuntu 20.04 (1.10.0 : january 2020). I didn't want to have to change the download url. It is a single binary.

It will allow to not use make when building. 

https://github.com/ninja-build/ninja/wiki/Pre-built-Ninja-packages